autolink/postlink/activityLinker.js

```diff
@@ -27,24 +56,32 @@ class ActivityLinker {
       debugn("   Removing getMainComponentName function");
       return contents.replace(/\/\*\*\s*\n([^\*]|(\*(?!\/)))*\*\/\s*@Override\s*protected\s*String\s*getMainComponentName\s*\(\)\s*{\s*return.+\s*\}/, "");
     }
+    warnn("   getMainComponentName function was not found.");
     return contents;
   }
 
   _extendNavigationActivity(activityContent) {
     if (this._doesActivityExtendReactActivity(activityContent)) {
-      debugn("   Extending NavigationActivity")
+      debugn("   Extending NavigationActivity");
       return activityContent
         .replace(/extends\s+ReactActivity\s*/, "extends NavigationActivity ")
-        .replace("import com.facebook.react.ReactActivity;", "import com.reactnativenavigation.NavigationActivity;")
-    } else {
-      warnn("   MainActivity already extends NavigationActivity")
+        .replace("import com.facebook.react.ReactActivity;", "import com.reactnativenavigation.NavigationActivity;");
     }
-    return activityContent;
+    if (this._hasAlreadyExtendReactActivity) {
+      warnn("   MainActivity already extends NavigationActivity");
+      return activityContent;
+    }
+
+    throw new Error("MainActivity was not successfully replaced. Please check the documentation and proceed manually.");
```

Comment:
This Error will be catched when the individual method step is called from the main link() method in the ActivityLinker Class.